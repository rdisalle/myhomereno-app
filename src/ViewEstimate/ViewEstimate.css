.ViewEstimatePage {
    margin-left: 10px;
    margin-right: 10px;
}

.EstimatePage_name {
    font-size: 20px;
    text-align: center;
}

@media only screen and (min-device-width: 768px) {
    .ViewEstimatePage {
        width: 600px;
        margin-right: auto;
        margin-left: auto;
    }
}

@media screen and (min-device-width: 1200px) {
    .ViewEstimatePage {
        width: 1000px;
        margin-right: auto;
        margin-left: auto;
    }
}