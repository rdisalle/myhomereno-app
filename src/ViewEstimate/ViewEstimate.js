import React, { Component } from 'react';
import './ViewEstimate.css';
import EstimatePage from '../EstimatePage/EstimatePage';
import MyHomeRenoContext from '../MyHomeRenoContext';
import PropTypes from 'prop-types';

class ViewEstimate extends Component {
  static defaultProps = {
    match: {
      params: {}
    }
  };

static contextType = MyHomeRenoContext;

onDelete = () => {
    const { estimates } = this.context;
    const { estimateId } = this.props.match.params;
    const estimate= this.findestimate(estimates, estimateId) || { content: '' };
    this.props.history.push(`/view-project-estimates/${estimate.project_id}`);
  }

findEstimate = (estimates=[], estimateId) =>
estimates.find(estimate => estimate.id === JSON.parse(estimateId));

  render() {
    const { estimates } = this.context;
    const { estimateId } = this.props.match.params;
    const estimate = this.findEstimate(estimates, estimateId) || { content: '' };

    return (
      <section className="ViewEstimatePage">
          <EstimatePage
              id={estimate.id}
              name={estimate.name}
              contractor_name={estimate.contractor_name}
              price={estimate.price}
              details={estimate.details}
              total_time={estimate.total_time}
              date_created={estimate.date_created}
              delete={this.onDelete}
          />
      </section>
  );
};
};

EstimatePage.propTypes = {
  match: PropTypes.object
};
  
export default ViewEstimate ;
