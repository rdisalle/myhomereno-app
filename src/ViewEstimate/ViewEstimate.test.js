import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ViewEstimate from './ViewEstimate'

describe(`ViewEstimate component`, () => {
  const props = {
    estimate: {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Estimate One",
        "project_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "contractor_name": "Mike's Plumbing",
        "price": "10,000",
        "details": "Can start next week",
        "total_time": "Four Weeks"
        },
  };

  it('renders a .ViewEstimate by default', () => {
    const wrapper = shallow(<ViewEstimate />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a estimate with estimate prop', () => {
    const estimate = shallow(<ViewEstimate {...props} />)
      .find('Estimate');
    expect(toJson(estimate)).toMatchSnapshot();
  });
});