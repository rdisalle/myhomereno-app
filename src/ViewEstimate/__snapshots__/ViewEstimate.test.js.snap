// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewEstimate component renders a .ViewEstimate by default 1`] = `
<section
  className="ViewEstimatePage"
>
  <EstimatePage
    contractor_name=""
    delete={[Function]}
    details=""
    name=""
    price=""
    total_time=""
  />
</section>
`;

exports[`ViewEstimate component renders a estimate with estimate prop 1`] = `null`;
